### PR TITLE
Set new HTTPS environment variable when using out of process

### PR DESCRIPTION
**Pull request by jkotalik**
Fixes https://github.com/aspnet/AspNetCore/issues/15243.

Today, ANCMV2 started setting the ASPNETCORE_HTTPS_PORT. This affected 2.2 applications that updated ANCMV2 to start always redirecting to HTTPS if the RedirectionMiddleware was enabled, as redirecting didn't work in 2.2 with ANCM.

This fix adds a new HTTPS_PORT environment variable called ANCM_HTTPS_PORT, which is now set by ANCM instead of HTTPS_PORT. The HttpsRedirectionMiddleware has also been updated to start reading this new value as a backup to the HTTPS_PORT. This means that if HTTPS_PORT is set, it will still always win over the ANCM port.

@BillHiebert this could affect IISExpress; when you set the HTTPS_PORT, do you set it to the HTTPS_PORT iisexpress is going to listen on? If so, then this change shouldn't break anything; the HTTPS_PORT still wins.

WIP because I still want to add another test with the HttpsRedirectionMiddleware.

@Pilchie for 3.1 approval

### CODE REVIEW

**Comment by Pilchie** [comment]
Scenario approved for 3.1.0-Preview3, assuming code reviews and CI are good before the cutoff.

File: src/Servers/IIS/IIS/test/testassets/InProcessWebSite/Startup.cs [hunk 1]
```diff
@@ -981,9 +981,16 @@ private async Task ProcessId(HttpContext context)
...
+        {
+            var httpsPort = context.RequestServices.GetService<IConfiguration>().GetValue<int?>("ANCM_HTTPS_PORT");
+
+            await context.Response.WriteAsync(httpsPort.HasValue ? httpsPort.Value.ToString() : "NOVALUE");
+        }
+
         public async Task HTTPS_PORT(HttpContext context)
         {
-            var httpsPort = context.RequestServices.GetService<IConfiguration>().GetValue<int?>("HTTPS_PORT");
+            var httpsPort = context.RequestServices.GetService<IConfiguration>().GetValue<int?>("ANCM_HTTPS_PORT");
```
Comment (Tratcher) [inline src/Servers/IIS/IIS/test/testassets/InProcessWebSite/Startup.cs]: Shouldn't this still be HTTPS_PORT?

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/HttpsTests.cs [hunk 2]
```diff
@@ -106,12 +106,32 @@ public async Task HttpsPortCanBeOverriden()
...
             var deploymentResult = await DeployAsync(deploymentParameters);
             var client = CreateNonValidatingClient(deploymentResult);
 
-            Assert.Equal("123", await client.GetStringAsync("/HTTPS_PORT"));
+            Assert.Equal("123", await client.GetStringAsync("/ANCM_HTTPS_PORT"));
+        }
+
+        [ConditionalFact]
+        [RequiresNewShim]
+        public async Task HttpsPortOldValueDoesNotRedirect()
```
Comment (Tratcher) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/HttpsTests.cs]:
```suggestion
        public async Task HttpsPortOldValueNotSet()
```

**Review by Tratcher — APPROVED** [review]

File: src/Servers/IIS/IIS/test/Common.FunctionalTests/HttpsTests.cs [hunk 3]
```diff
@@ -106,12 +106,32 @@ public async Task HttpsPortCanBeOverriden()
...
+                    element.Descendants("bindings")
+                        .Single()
+                        .GetOrAdd("binding", "protocol", "https")
+                        .SetAttributeValue("bindingInformation", $":{TestPortHelper.GetNextSSLPort()}:localhost");
+                });
+
+            var deploymentResult = await DeployAsync(deploymentParameters);
+            var client = CreateNonValidatingClient(deploymentResult);
+
+            Assert.Equal("NOVALUE", await client.GetStringAsync("/HTTPS_PORT"));
```
Comment (halter73) [inline src/Servers/IIS/IIS/test/Common.FunctionalTests/HttpsTests.cs]: Does ANCM_HTTPS_PORT have a value set in this test? Either way, can we verify that the is/isn't set in this test?

Comment (halter73) [reply src/Servers/IIS/IIS/test/Common.FunctionalTests/HttpsTests.cs]: My assumption is that ANCM_HTTPS_PORT should equal whatever TestPortHelper.GetNextSSLPort() returned, right?

Comment (jkotalik) [reply src/Servers/IIS/IIS/test/Common.FunctionalTests/HttpsTests.cs]: Yeah, this was already done by another test. I now consolidated them.

**Comment by jkotalik** [comment]
@BrennanConroy TransportCommunicatesErrorToApplicationWhenClientDisconnectsAbnormally (SignalR test), known flaky? https://dev.azure.com/dnceng/public/_build/results?buildId=411560&view=ms.vss-test-web.build-test-results-tab&runId=12900532&resultId=105256&paneView=debug

**Comment by BrennanConroy** [comment]
>known flaky?

😢 no

**Pull request merged**